src/main/java/io/github/foundationgames/jsonem/serialization/JsonEMCodecs.java

```diff
@@ -18,12 +18,21 @@
 import net.minecraft.client.model.TexturedModelData;
 import net.minecraft.client.util.math.Vector2f;
 import net.minecraft.util.Util;
-import net.minecraft.util.math.Vec3f;
+
+import org.joml.Vector3f;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Optional;
 
 public class JsonEMCodecs {
+        private static final Codec<Vector3f> VECTOR3F_CODEC = Codec.FLOAT.listOf().comapFlatMap(
```

Comment:
- Should be named `VECTOR3F`, and placed above the Vector2f codec
- Incorrect indentation

Reply:
it has to appear before the other `public static final` codecs so I'll move those below vector2f I guess

Reply:
Nothing *has* to be in any specific order, you can put it right at the end of the file if you wanted to.

Reply:
Yeah, the ordering is purely for convention, to make it easier to find specific codecs for things

Reply:
> Nothing _has_ to be in any specific order, you can put it right at the end of the file if you wanted to.

![image](https://user-images.githubusercontent.com/4142480/229678378-71d4714d-6c6e-4291-b702-39ac3d48eef4.png)

🤷 guess ordering isn't purely for convention